### PR TITLE
hotfix security agent: only refresh state when connected to new relic

### PR DESCRIPTION
**Pull request by iamemilio**

### CODE REVIEW

File: v3/newrelic/secure_agent.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ func (app *Application) RegisterSecurityAgent(s securityAgent) {
 	if app != nil && app.app != nil && s != nil {
 		secureAgent = s
 		run, _ := app.app.getState()
-		if run != nil {
+		if run.Reply.IsConnectedToNewRelic() {
```
Comment (nr-swilloughby) [inline v3/newrelic/secure_agent.go:46]: Is there any case where `run` would be valid but `run.Reply` be `nil`? just wondering if we should make that `if run.Reply != nil && run.Reply.IsConnectedToNewRelic()`

**Review by nr-swilloughby — CHANGES_REQUESTED** [review]

File: v3/newrelic/secure_agent.go (hunk 1 above)
Comment (iamemilio) [reply v3/newrelic/secure_agent.go:46]: As the code is written now, we will always have either a real run.Reply, or a placeholder. We can check for to be defensive against future changes maybe?

**Review by mirackara — APPROVED** [review]

File: v3/newrelic/secure_agent.go (hunk 1 above)
Comment (nr-swilloughby) [reply v3/newrelic/secure_agent.go:46]: probably wise

Comment (iamemilio) [reply v3/newrelic/secure_agent.go:46]: added

**Review by nr-swilloughby — APPROVED** [review]

**Pull request merged**